src/Common/src/Interop/Unix/System.Native/Interop.Read.cs

```diff
@@ -19,7 +20,11 @@ internal static partial class Sys
         /// Returns the number of bytes read on success; otherwise, -1 is returned
         /// Note - on fail. the position of the stream may change depending on the platform; consult man 2 read for more info
         /// </returns>
+#if MONO
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
```

Comment:
We do not want to pass `SafeFileHandle` to native without doing the marshalling in managed, so it would be much easier to implement our own implementation of `Interop.Sys.Read` without relying on the `Interop.Read.cs` file. Please add a `mcs/class/corlib/corefx/Interop.Read.cs` file with the appropriate implementation of `Interop.Sys.Read`